jsonclient/client.go

```diff
@@ -41,6 +44,69 @@ type JSONClient struct {
 	httpClient *http.Client          // used to interact with the server via HTTP
 	Verifier   *ct.SignatureVerifier // nil for no verification (e.g. no public key available)
 	logger     Logger                // interface to use for logging warnings and errors
+
+	multiplier int64
+	until      *time.Time
+	mu         sync.RWMutex
+}
+
+const (
+	maxBackoff = time.Second * 128
+	jitter     = 250
+)
+
+func (c *JSONClient) setBackoff(override *time.Duration) time.Duration {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.until != nil {
+		return time.Until(*c.until)
+	}
+	var wait time.Duration
+	if override != nil {
+		wait = *override
+	} else {
+		c.multiplier++
```

Comment:
Having no limit for this multiplier feels worrying, particularly for any long-lived client instances now that backoff state is persistent.  A couple of hours of backoffs pegged at 128s  will get the multiplier up to ~55, which will take a long time to get back down to 1.